### ISSUE TITLE
keras Batch Norm Issure

### ISSUE THREAD

**Issue by sunshinenum** [tensorflow/recommenders-addons#326]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos7
- TensorFlow version and how it was installed (source or binary): 2.8.3
- TensorFlow-Recommenders-Addons version and how it was installed (source or binary): 0.6.0
- Python version: 3.9
- Is GPU used? (yes/no): no

**Describe the bug**

In predict_step, if I set training=False, I would get a very low auc. It seems that, the params of BN in keras were not saved during model save operation.

**Code to reproduce the issue**

```
## evaluate
y_pred_ctr, bias = self(features, training=False)
## load model
self.model.load_weights(os.path.join(self.model_dir, 'weights'))
## save model
save_options = tf.saved_model.SaveOptions(namespace_whitelist=['TFRA'])
self.model.save_weights(os.path.join(self.model_dir, 'weights'), options=save_options)
```

Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by MoFHeka** [comment]
@sunshinenum This issue is solved for the time being, I will close this issue. Turn it back on if necessary.
